elasticsearch-agent/src/main/java/com/epam/pipeline/elasticsearchagent/service/impl/NFSSynchronizer.java

```diff
@@ -107,15 +109,19 @@ public void synchronize(final LocalDateTime lastSyncTime, final LocalDateTime sy
         log.debug("Started NFS synchronization");
         fileMapper.updateSearchMasks(cloudPipelineAPIClient, log);
 
-        List<AbstractDataStorage> allDataStorages = cloudPipelineAPIClient.loadAllDataStorages();
-        allDataStorages.stream()
-                .filter(dataStorage -> dataStorage.getType() == DataStorageType.NFS)
+        cloudPipelineAPIClient.loadAllDataStoragesWithMounts().stream()
+                .filter(dataStorage -> dataStorage.getStorage().getType() == DataStorageType.NFS)
                 .forEach(this::createIndexAndDocuments);
     }
 
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    void createIndexAndDocuments(final AbstractDataStorage dataStorage) {
-        EntityPermissionVO entityPermission = cloudPipelineAPIClient
+    void createIndexAndDocuments(final DataStorageWithShareMount storageWithShareMount) {
+        final AbstractDataStorage dataStorage = storageWithShareMount.getStorage();
+        log.debug("Starting to  process storage: {}, id: {}.", dataStorage.getName(), dataStorage.getId());
+        final String regionCode = Optional.ofNullable(storageWithShareMount.getShareMount())
```

Comment:
I think we should preload all regions rather than call region api for each storage